code/modules/mob/mob_movement.dm

```diff
@@ -152,146 +157,135 @@
+	if(moving)
+		return 0
 
 	if(isliving(mob))
 		var/mob/living/L = mob
 		if(L.incorporeal_move)//Move though walls
 			Process_Incorpmove(direct)
 			return
 
-	if(Process_Grab())	return
-
-	if(mob.buckled)							//if we're buckled to something, tell it we moved.
-		return mob.buckled.relaymove(mob, direct)
-
-	if(mob.remote_control)					//we're controlling something, our movement is relayed to it
+	if(mob.remote_control) //we're controlling something, our movement is relayed to it
 		return mob.remote_control.relaymove(mob, direct)
 
 	if(isAI(mob))
 		if(istype(mob.loc, /obj/item/device/aicard))
 			var/obj/O = mob.loc
-			return O.relaymove(mob, direct) //aicards have special relaymove stuff
-		return AIMove(n,direct,mob)
+			return O.relaymove(mob, direct) // aicards have special relaymove stuff
+		return AIMove(n, direct, mob)
 
-	if(!mob.canmove)
+
+	if(Process_Grab())
 		return
 
-	//if(istype(mob.loc, /turf/space) || (mob.flags & NOGRAV))
-	//	if(!mob.Process_Spacemove(0))	return 0
+	if(mob.buckled) //if we're buckled to something, tell it we moved.
+		return mob.buckled.relaymove(mob, direct)
+
+	if(!mob.canmove)
+		return
 
 	if(!mob.lastarea)
 		mob.lastarea = get_area(mob.loc)
 
-	if(isobj(mob.loc) || ismob(mob.loc))//Inside an object, tell it we moved
+	if(isobj(mob.loc) || ismob(mob.loc)) //Inside an object, tell it we moved
 		var/atom/O = mob.loc
 		return O.relaymove(mob, direct)
+
+	if(!mob.Process_Spacemove(direct))
+		return 0
+
+	if(mob.restrained()) // Why being pulled while cuffed prevents you from moving
+		for(var/mob/M in orange(1, mob))
+			if(M.pulling == mob)
+				if(!M.incapacitated() && mob.Adjacent(M))
+					to_chat(src, "<span class='warning'>You're restrained! You can't move!</span>")
+					move_delay = world.time + 10
+					return 0
+				else
+					M.stop_pulling()
+
+
+	//We are now going to move
+	moving = 1
+	move_delay = mob.movement_delay() + world.time
+	mob.last_movement = world.time
+
+	if(locate(/obj/item/weapon/grab, mob))
+		move_delay = max(move_delay, world.time + 7)
+		var/list/L = mob.ret_grab()
+		if(istype(L, /list))
+			if(L.len == 2)
+				L -= mob
+				var/mob/M = L[1]
+				if(M)
+					if((get_dist(mob, M) <= 1 || M.loc == mob.loc))
+						var/turf/prev_loc = mob.loc
+						. = ..()
+						if(M && isturf(M.loc)) // Mob may get deleted during parent call
+							var/diag = get_dir(mob, M)
+							if((diag - 1) & diag)
+							else
+								diag = null
+							if((get_dist(mob, M) > 1 || diag))
+								step(M, get_dir(M.loc, prev_loc))
+			else
+				for(var/mob/M in L)
+					M.other_mobs = 1
+					if(mob != M)
+						M.animate_movement = 3
+				for(var/mob/M in L)
+					spawn(0)
+						step(M, direct)
+						return
+					spawn(1)
+						M.other_mobs = null
+						M.animate_movement = 2
+						return
+
+	else if(mob.confused)
+		step(mob, pick(cardinal))
+	else
+		. = ..()
 
 	if(istype(mob.get_active_hand(), /obj/item))

```

Comment:
Ahh, this is for the cable layer; if you implemented TG's `on_mob_move` system, this could probably be refactored. I'll leave it up to you whether or not you want to do that in this PR though.